### DISCUSSION TITLE
IR Watchdog : Tobii Dynavox : Not Failing Enough

### DISCUSSION THREAD

**Discussion by nelsonii** [ATMakersOrg/EyeAlert#17] [Q&A]
Here's what I'm seeing with the Tobii Dynavox I-15:

When it locks up / becomes unusable / reboots the IR emitters still run. This is bad news for the IR Watchdog, as its looking for IR -- and the IR is there -- but the OS is stuck rebooting ("your windows machines did not shut down properly...").

Why is this? I believe that the IR emitters and cameras are all on their own board / separate from the PC portion. Think of it as a Raspberry Pi and Windows PC in the same enclosure, talking to each other. 

This makes perfect sense from a design standpoint (they bolt-on the eye tracking to off-the-shelf hardware, in a custom enclosure). Long story short, the IR / camera logic board is humming right along -- but the Windows PC fails.

There are management tools out there to watch out for system (OS) failures / reboots / etc. But most are designed for data center use (when you're monitoring thousands of servers). The software is often called "Operations Managers". Perhaps we can build something like that, but with no installs. Something I'm pondering (look for USB activity? look for something on the screen? etc).

We'll have to see how the other brands/version of eye tracking behave.

**Comment by nelsonii** [comment]
During on call on Wednesday, 2022-06-01, Nachum mentioned that he's aware of the Tobii behavior and that we should continue moving forward.

**Reply by JayToltTech** [reply]
Yes, IR emitters glowing (or not) are sometimes but not always associated with eye tracking failure.
